V_pk_fma

amcamd opened the pull request:
Changes to get hgemm to call 32 bit vectorized_packed_fma in place of 16 bit multiply and add. Note that this pull request has no changes for load and store commands. The plan is to address these in assembly code, not with half4 or half8 loads.

There is a change to initialize matrices with random values in [-max,max] rather than values in [0,max]. The reason for this change is to reduce the accumulator value in the reduction sums. This is achieved by having negative and positive values in the reduction sum. This is required because the 16 bit reduction sums for matrix size 4096 were giving overflow and underflow to inf or -inf.

newling commented:
@amcamd @guacamoleo 
I've noticed that MIOpenGEMM kernels run slower when buffers are initialized with random values in [-R, R], as opposed to [0, R]. Maybe something to check out for in Tensile? I don't know what type of compression is being done in the background for this to happen, I'd be interested if you have any idea.

amcamd commented:
Addressed all comments with 4 commits:

1. Moved few lines of code from KernelWriter.py to KernelWriterSource.py
2. Renamed TYPE_MAC_HALF2 to TYPE_MAC, and conditionally defined TYPE_MAC
3. Rearrange loops to minimize code divergence
4. Move half declarations from TensileTypes.h to KernelHeader.h

amcamd commented:
Hi David,

I can confirm that tests passed for the following .yaml files

test_hgemm.yaml  :   hip, (OCL cannot be run because no half type in OCL)

test_sgemm.yaml  :  both hip and OCL

test_dgemm_defaults.yaml  :  both hip and OCL

Thanks,
Andrew
From: David Tanner [mailto:notifications@github.com]
Sent: Monday, October 2, 2017 12:20 PM
To: ROCmSoftwarePlatform/Tensile <Tensile@noreply.github.com>
Cc: Chapman, Andrew <Andrew.Chapman@amd.com>; Mention <mention@noreply.github.com>
Subject: Re: [ROCmSoftwarePlatform/Tensile] V_pk_fma (#114)


Merged #114<https://github.com/ROCmSoftwarePlatform/Tensile/pull/114>.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/ROCmSoftwarePlatform/Tensile/pull/114#event-1274466639>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AX2-1n-x-aBfQSf3p1hnny3jkmGWzgwZks5soRtEgaJpZM4PnkXc>.
